packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Providers/ModelProvider.cs

```diff
@@ -361,12 +363,21 @@ private Dictionary<InputModelType, Dictionary<string, InputModelProperty>> Build
             return derivedProperties;
         }
 
+        private void EnumerateDerivedModels(InputModelType inputModel, List<InputModelType> derivedModels)
+        {
+            foreach (var derivedModel in inputModel.DerivedModels)
+            {
+                derivedModels.Add(derivedModel);
+                EnumerateDerivedModels(derivedModel, derivedModels);
+            }
+        }
+
         protected override PropertyProvider[] BuildProperties()
         {
             var propertiesCount = _inputModel.Properties.Count;
             var properties = new List<PropertyProvider>(propertiesCount + 1);
 
-            Dictionary<string, InputModelProperty> baseProperties = _inputModel.BaseModel?.Properties.ToDictionary(p => p.Name) ?? [];
+            Dictionary<string, InputModelProperty> baseProperties = EnumerateBaseModels().SelectMany(m => m.Properties).GroupBy(x => x.Name).Select(g => g.First()).ToDictionary(p => p.Name) ?? [];
```

Comment:
I don't think we need the GroupBy. There shouldn't be duplicate named properties.

Reply:
There was a failing test with duplicated named properties.
https://dev.azure.com/azure-sdk/public/_build/results?buildId=4780341&view=logs&j=ed251b61-5f89-5e2e-e553-4308b1094c74&t=3be794ed-f976-5e6a-a6f6-7dac6e90dcfe&l=1126

Reply:
I think it is possible to have duplicates here.
Imagine we have derivedModel2 -> derivedModel1 {"prop1": ...} -> baseModel {"prop1":..}
property of same name can be overridden in derivedModel1

Reply:
We need the groupby, it is failing on a spec test without grouping [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=4784784&view=logs&j=5913cd1b-7ca6-53ba-f6fc-24af8be2f360&t=b343c40a-34f1-5b24-c106-19474c641250&l=3396)

Reply:
Thanks for investigating this!